sia/hostdb/hostdb.go

```diff
@@ -84,14 +114,16 @@ func (hdb *HostDatabase) Update(initialStateHeight consensus.BlockHeight, rewoun
 
 	// Add hosts found in blocks that were applied.
 	for _, b := range appliedBlocks {
-		var entries []HostEntry
+		var entries []components.HostEntry
 		entries, err = findHostAnnouncements(initialStateHeight, b)
 		if err != nil {
 			return
 		}
 
 		for _, entry := range entries {
+			hdb.unlock()
 			err = hdb.Insert(entry)
+			hdb.lock()
```

Comment:
mutex convention makes this kinda ugly. You could have an `insert` function that was `Insert` without the locks, and then define `Insert` as:
```
func (hdb *HostDB) Insert(entry) error {
    hdb.lock()
    defer hdb.unlock()
    return hdb.insert(entry)
}
```